src/txmempool.cpp

```diff
@@ -305,19 +305,19 @@ void CTxMemPool::UpdateForRemoveFromMempool(const setEntries &entriesToRemove, b
 
 void CTxMemPoolEntry::UpdateDescendantState(int64_t modifySize, CAmount modifyFee, int64_t modifyCount)
 {
-    nSizeWithDescendants += modifySize;
+    nSizeWithDescendants = (uint64_t)((int64_t)nSizeWithDescendants + modifySize);
```

Comment:
could you explain what you're doing here?
I get the impression that you intend to make it so that (unintended) unsigned wraparound does not happen. I don't think it's obvious how these casts help with that. is `modifySize` legitimately negative sometimes? and the `nSizeWithDescendants` is legitimately near `INT64_MAX`?

Reply:
@arvidn Yes `modifySize` can be legitimately negative. That condition and the resulting wraparound is trigged just by running the test suite:

```
./configure --with-sanitizers=integer && make check && test/functional/test_runner.py
```